src/lib/support/BytesToHex.h

```diff
@@ -126,23 +188,75 @@ inline CHIP_ERROR BytesToLowercaseHexString(const uint8_t * src_bytes, size_t sr
 }
 
 /**
- * Convert a buffer of hexadecimal characters to bytes.  Supports both lowercase
+ * Convert a buffer of hexadecimal characters to bytes. Supports both lowercase
  * and uppercase (or a mix of cases) hexadecimal characters. Supported input is
- * [0-9a-fA-F].
+ * [0-9a-fA-F]. The input is assumed to be in a big-endian order. The outpu is
+ * also in a big-endian order.
+ *
+ * @param src_hex a pointer to the character buffer to convert. It is not
+ *                assumed to be null-terminated.
+ * @param src_size the length of the character buffer to convert.
+ * @param dest_bytes the buffer to fill with the decoded bytes.
+ * @param dest_size_max the total size of the buffer to be filled.
+ *
+ * @return 0 on errors:
+ *           - dest_size_max not big enough.
+ *           - src_size not even.
+ *           - Some character not in [0-9a-fA-F] is present in src_hex.
+ *         number of bytes actually decoded from the string on success.
+ */
+size_t HexToBytes(const char * src_hex, const size_t src_size, uint8_t * dest_bytes, size_t dest_size_max);
+
+/**
+ * Convert a buffer of hexadecimal characters into uin64_t. Supports only
+ * uppercase hexadecimal input characters. Supported input is [0-9A-F].
+ * The input is assumed to be in a big-endian order.
+ *
+ * @param src_hex a pointer to the character buffer to convert. It is not
+ *                assumed to be null-terminated.
+ * @param src_size the length of the character buffer to convert.
+ * @param dest 64-bit number to output.
+ *
+ * @return 0 on errors:
+ *           - src_size not even.
+ *           - Some character not in [0-9A-F] is present in src_hex.
+ *         number of bytes actually decoded from the string on success.
+ */
+size_t UppercaseHexToBytes(const char * src_hex, const size_t src_size, uint64_t & dest);
+
+/**
```

Comment:
Suggest referring between uint32/16 versions back to uint64 version

Reply:
done